hometask-4-desing-patterns/shipment-state/shipment-state.ts

```diff
@@ -0,0 +1,77 @@
+import { PackageType } from "../enums/package-type.enum";
+import { Shipper } from "./shipper/shipper";
+import { capitalizeFirstLetter } from '../helpers'
+import { Mark } from "../enums";
+import { airEastShipper, chicagoSprintShipper, pacificParcelShipper } from "./shipper";
+
+export class ShipmentSate {
+	private _shipper: Shipper;
+	private _packageType: PackageType;
+	private _shipmentId: number;
+	private _weight: number;
+	private _senderAdress: string;
+	private _senderZipCode: string;
+	private _deliveryAdress: string;
+	private _deliveryZipCode: string;
+	private _mark: Mark | null;
+
+	public set packageType(value: PackageType) {
+		this._packageType = value
+	}
+
+	public set weight(value: number) {
+		this._weight = value
+	}
+
+	public set senderAdress(value: string) {
+		this._senderAdress = value
+	}
+
+	public set senderZipCode(value: string) {
+		this._senderZipCode = value
+		if (value[0] === '1' || '2' || '3') {
+			this._shipper = airEastShipper
```

Comment:
could you please implement your strategy as described here with context? https://refactoring.guru/design-patterns/strategy/typescript/example